frontend/src/libraries/config.ts

```diff
@@ -1,14 +1,29 @@
+import { IncomingMessage } from "http";
 import { ExplorerConfig, NearNetwork } from "next.config";
 import getConfig from "next/config";
 
 const {
   publicRuntimeConfig: { nearNetworks, nearNetworkAliases },
 } = getConfig() as ExplorerConfig;
 
-export function getNearNetwork(hostname?: string): NearNetwork {
+function getHostname(req?: IncomingMessage): string | undefined {
+  if (typeof window !== "undefined") {
+    return window.location.host;
+  } else if (req) {
+    return req.headers.host;
+  }
+}
+
+export function getNearNetwork(req?: IncomingMessage): NearNetwork {
```

Comment:
Actually, we have only two options on this function usage:
- we use it server-side hence we have `req`
- we use it client-side hence we have `window.location.host`

Reply:
I am not sold on having "get near network by incoming message". The previous implementation with "get near network by hostname" makes more sense to me.

Reply:
It does make sense, but also makes us duplicate extracting this hostname (not only from `req`, but also from `window.location`).
We can keep function dependent on `hostname`, but then we need another one to extract `hostname` from `req` / `window`.

What's your suggest?